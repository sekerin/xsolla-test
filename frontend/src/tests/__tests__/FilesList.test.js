import React from 'react';

import { shallow }       from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import FileList       from '../../components/FilesList/FilesList';
import configureStore from '../../redux/configureStore';

describe('Test FilesList component', () => {
  it('Empty snapshot', () => {
    const store = configureStore({ filesList: { items: [], modal: false } });
    const wrapper = shallow(<FileList store={store}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('FileList with data', () => {
    const store = configureStore({ filesList: { items: [{ name: 'fileName' }], modal: false } });
    const wrapper = shallow(<FileList store={store}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
