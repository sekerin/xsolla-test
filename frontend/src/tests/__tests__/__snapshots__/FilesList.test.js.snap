// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FilesList component Empty snapshot 1`] = `
<FileList
  dispatch={[Function]}
  items={Array []}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "liftedStore": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "liftedStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
